packages/affiliates/libs/uniswap/models.ts

```diff
@@ -144,36 +118,50 @@ export const Pools = () => {
 };
 export type Pools = ReturnType<typeof Pools>;
 
-export type Tick = {
-  // the total position liquidity that references this tick
-  liquidityGross?: string;
-  // amount of net liquidity added (subtracted) when tick is crossed from left to right (right to left),
-  liquidityNet?: string;
-  // fee growth per unit of liquidity on the _other_ side of this tick (relative to the current tick)
-  // only has relative meaning, not absolute — the value depends on when the tick is initialized
-  feeGrowthOutside0X128?: string;
-  feeGrowthOutside1X128?: string;
-  // the cumulative tick value on the other side of the tick
-  tickCumulativeOutside?: string;
-  // the seconds per unit of liquidity on the _other_ side of this tick (relative to the current tick)
-  // only has relative meaning, not absolute — the value depends on when the tick is initialized
-  secondsPerLiquidityOutsideX128?: string;
-  // the seconds spent on the other side of the tick (relative to the current tick)
-  // only has relative meaning, not absolute — the value depends on when the tick is initialized
-  secondsOutside?: string;
-  // true iff the tick is initialized, i.e. the value is exactly equivalent to the expression liquidityGross != 0
-  // these 8 bits are set to prevent fresh sstores when crossing newly initialized ticks
-  initialized?: boolean;
-  pool?: string;
-  index?: string;
-  id?: string;
-};
-export const Ticks = () => {
-  function makeId(state: Tick) {
-    const { pool, index } = state;
-    return [pool, index].join("!");
+export const Balances = () => {
+  let total = 0n;
+  const store = new Map<Id, string>();
+  function getOrCreate(address: string) {
+    if (store.has(address)) return store.get(address);
+    return create(address);
+  }
+  function create(addr: string) {
+    assert(!store.has(addr), "Already has address");
+    store.set(addr, "0");
+    return "0";
+  }
+  function add(addr: string, amount: string) {
+    const amountn = BigInt(amount);
+    assert(amountn >= 0n, "amount must be >= 0: " + amount);
+    const balance = getOrCreate(addr);
+    const result = (BigInt(balance) + amountn).toString();
+    store.set(addr, result);
+    total = total + amountn;
```

Comment:
Personally, I agree that BigInts are just better than BNs since you can use all the normal primitive syntax. This is especially convenient for comparison operators.